File: MessageUtils/ActionBarBuilder.java
package net.glitchtechs.starapi.MessageUtils;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

import net.md_5.bungee.api.ChatMessageType;
import net.md_5.bungee.api.chat.TextComponent;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.plugin.java.JavaPlugin;

public class ActionBarBuilder {
    private final JavaPlugin plugin;
    private final Map<UUID, Integer> actionBars;
    private String message;

    public ActionBarBuilder(JavaPlugin plugin) {
        this.plugin = plugin;
        this.actionBars = new HashMap();
    }

    public ActionBarBuilder message(String message) {
        this.message = MessageColors.hex(message);
        return this;
    }

    public void send(Player player) {
        int taskId = Bukkit.getScheduler().scheduleSyncDelayedTask(this.plugin, () -> {
            player.spigot().sendMessage(ChatMessageType.ACTION_BAR, TextComponent.fromLegacyText(this.message));
            this.actionBars.remove(player.getUniqueId());
        });
        this.actionBars.put(player.getUniqueId(), taskId);
    }

    public void clear(Player player) {
        Integer taskId = (Integer)this.actionBars.remove(player.getUniqueId());
        if (taskId != null) {
            Bukkit.getScheduler().cancelTask(taskId);
        }

    }
}

File: Kotlin/src/ActionBarBuilder.kt
package net.glitchtechs.starapi

import net.md_5.bungee.api.ChatMessageType
import net.md_5.bungee.api.chat.TextComponent
import org.bukkit.Bukkit
import org.bukkit.entity.Player
import org.bukkit.plugin.java.JavaPlugin
import java.util.*


class ActionBarBuilder(private val plugin: JavaPlugin) {
    private val actionBars: MutableMap<UUID?, Int?>
    private var message: String? = null

    init {
        this.actionBars = HashMap()
    }

    fun message(message: String?): ActionBarBuilder {
        this.message = MessageUtils.hex(message!!)
        return this
    }

    fun send(player: Player) {
        val taskId = Bukkit.getScheduler().scheduleSyncDelayedTask(this.plugin) {
            player.spigot().sendMessage(ChatMessageType.ACTION_BAR, *TextComponent.fromLegacyText(this.message))
            actionBars.remove(player.uniqueId)
        }
        actionBars[player.uniqueId] = taskId
    }

    fun clear(player: Player) {
        val taskId = actionBars.remove(player.uniqueId)
        if (taskId != null) {
            Bukkit.getScheduler().cancelTask(taskId)
        }
    }
}